pandas/core/nanops.py

```diff
@@ -254,7 +254,7 @@ def nansum(values, axis=None, skipna=True):
 @bottleneck_switch()
 def nanmean(values, axis=None, skipna=True):
     values, mask, dtype, dtype_max = _get_values(values, skipna, 0)
-    the_sum = _ensure_numeric(values.sum(axis, dtype=dtype_max))
+    the_sum = _ensure_numeric(values.sum(axis, dtype=np.float64))
```

Comment:
oh ... you are right. This means we actually have more to fix than the cases for int types. Because `dtype_max` would be `float64` even when the inputs are float32... I'll update shortly.